### DISCUSSION TITLE
Insert datetime using the "Timenow" plugin?

### DISCUSSION THREAD

**Discussion by zohozer** [facelessuser/RegReplace#110] [Q&A]
Hi all. If I understand correctly RegReplace can use external plugins for various tasks.

My question is how I can use the [Timenow](https://packagecontrol.io/packages/Timenow) to insert the datetime that can be found in the Edit/Timenow/Insert datetime from Sublime Text menu after installing the package.

I do need to insert the date and time at which I performed a RegReplace command like:

`Date: 2024-10-15 14:15`

Thank you.

**Comment by facelessuser** [comment]
I don't use Timenow, so I don't know how it works or how you could tap into it. You would have to investigate that yourself.

As far as doing custom things, you could likely utilize the replace plugin to do what you need. The documentation describes this and shows and example: https://facelessuser.github.io/RegReplace/usage/#custom-replace-plugins.

**Comment by zohozer** [comment]
I took a look at the documentation in the link provided, but unfortunately there it is an example of incrementing a date that it is found in the document. I am not a programmer, I do know only some very basic Python and I don't understand how I can use this example to insert the current date/time into my text files. I will appreciate if you can provide-me something that I can actually use. Thank you for understanding.

P.S. I found some documentation on how to get the NOW date/time in Python, but still I don't know how I can create myself an usable plugin from this:

[Getting today's date in YYYY-MM-DD in Python?](https://stackoverflow.com/questions/32490629/getting-todays-date-in-yyyy-mm-dd-in-python)

**Comment by zohozer** [comment]
Ok, I made some progress, but still need a little bit of help.

I created a new module called "TimeNow.py" and placed into "Packages\User" folder.

```
# https://www.reddit.com/r/SublimeText/comments/5p4to8/insert_datetime_sublime_text_notepad_f5/
import sublime, sublime_plugin, time
class InsertDatetimeCommand(sublime_plugin.TextCommand):
    def run(self, edit):
        sel = self.view.sel();
        for s in sel:
            self.view.insert(edit, s.end(), time.strftime( ' %H:%M:%S %d/%m/%Y ' ))
```

I assigned a key binding in "Preferences/keybindings/User":

`{ "keys": ["f5"], "command": "insert_datetime"}`

I tested and it is inserting the current Date/Time in document when I do press the F5 key.

Now, the only thing missing it is how I can invoke this command in RegReplace to insert the current Date/Time in my files?

**Comment by zohozer** [comment]
Almost there, still get an error:

```
reloading settings Packages/User/reg_replace_rules.sublime-settings
Traceback (most recent call last):
  File "C:\Users\Program\AppData\Roaming\Sublime Text\Installed Packages\RegReplace.sublime-package\rr_replacer.py", line 132, in on_replace
    text = module.replace(m, **self.plugin_args)
AttributeError: 'module' object has no attribute 'replace'
```

I placed the "TimeNow.py" in the "Sublime Text\Packages" folder and created the following reg_replace_rule:

```
      "Insert_Date": {
      "find": "InsertTimeNowHere",
      "plugin": "TimeNow"
    },
```

I am invoking my command with:

```
"args": {
      "replacements": [
        "r130_Header",
        "Insert_Date",
       ]
```

but unfortunately I do get the above error.

**Reply by facelessuser** [reply]
>  I am not a programmer,

I don't have time to teach programming or to code custom solutions for everyone who has questions. Please read the documentation and start by using the example. Get that working first so you have a working example.

The documentation is very clear that you load a module, and it looks for the `replace` function in that module.  That is why you are getting the error.

```
AttributeError: 'module' object has no attribute 'replace'
```

The example is given so that people can try, test it, and then change it to suite there needs, removing anything that no longer applies to their solution.

You cannot give RegReplace a random module, it needs to be formatted per RegReplace's expectations. Your `replace` function can return something like `time.strftime( ' %H:%M:%S %d/%m/%Y ' )` if that is what you want.
